Trimble.Modus.Components/Controls/Layouts/WrapLayout.cs

```diff
@@ -0,0 +1,252 @@
+using Microsoft.Maui.Layouts;
+using MIView = Microsoft.Maui.IView;
+
+namespace Trimble.Modus.Components.Controls.Layouts;
+
+/// <summary>
+/// A layout that arranges views in a wrapping grid.
+/// Ref: https://github.com/mdc-maui/mdc-maui/blob/master/Material.Components.Maui/Components/Layouts/WrapLayout.cs
+/// </summary>
+public partial class WrapLayout : Layout, IItemsLayout
+{
+
+    public static readonly BindableProperty OrientationProperty =
+     BindableProperty.Create(
+         nameof(Orientation),
+         typeof(StackOrientation),
+         typeof(WrapLayout),
+         StackOrientation.Horizontal,
+         propertyChanged: OnPropertyChanged);
+
+    public static readonly BindableProperty SpacingProperty =
+     BindableProperty.Create(
+         nameof(Spacing),
+         typeof(double),
+         typeof(WrapLayout),
+         0d,
+         propertyChanged: OnSpacingChanged);
+
+    public static readonly BindableProperty HorizontalSpacingProperty =
+     BindableProperty.Create(
+         nameof(HorizontalSpacing),
+         typeof(double),
+         typeof(WrapLayout),
+         0d,
+         propertyChanged: OnPropertyChanged);
+
+    public static readonly BindableProperty VerticalSpacingProperty =
+     BindableProperty.Create(
+         nameof(VerticalSpacing),
+         typeof(double),
+         typeof(WrapLayout),
+         0d,
+         propertyChanged: OnPropertyChanged);
+
+    private bool isOnSpacingChanging;
+
+    public StackOrientation Orientation
+    {
+        get => (StackOrientation)GetValue(OrientationProperty);
+        set => SetValue(OrientationProperty, value);
+    }
+    public double Spacing
+    {
+        get => (double)GetValue(SpacingProperty);
+        set => SetValue(SpacingProperty, value);
+    }
+
+    public double HorizontalSpacing
+    {
+        get => (double)GetValue(HorizontalSpacingProperty);
+        set => SetValue(HorizontalSpacingProperty, value);
+    }
+
+    public double VerticalSpacing
+    {
+        get => (double)GetValue(VerticalSpacingProperty);
+        set => SetValue(VerticalSpacingProperty, value);
+    }
+
+    private static void OnSpacingChanged(BindableObject bindable, object oldValue, object newValue)
```

Comment:
Add regions and follow the regions layer

Reply:
Done